File: packages/sdk-ts/src/resources/knowledge-bases.ts
import type { Http, RequestOptions } from '../core/http.js';
import type { Page } from '../core/page.js';
import type { CreateKnowledgeBase, KnowledgeBase, PaginationQuery, UpdateKnowledgeBase } from '../types.js';

const enc = encodeURIComponent;

/** Knowledge bases: facts and FAQs an agent can draw on. */
export class KnowledgeBases {
    constructor(private readonly http: Http) {}

    list(query: PaginationQuery = {}, opts?: RequestOptions): Promise<Page<KnowledgeBase>> {
        return this.http.list<KnowledgeBase>('/v1/knowledge-bases', query, opts);
    }

    create(body: CreateKnowledgeBase, opts?: RequestOptions): Promise<KnowledgeBase> {
        return this.http.request<KnowledgeBase>('POST', '/v1/knowledge-bases', { ...opts, body });
    }

    retrieve(id: string, opts?: RequestOptions): Promise<KnowledgeBase> {
        return this.http.request<KnowledgeBase>('GET', `/v1/knowledge-bases/${enc(id)}`, opts);
    }

    /** Only the fields sent are changed. */
    update(id: string, body: UpdateKnowledgeBase, opts?: RequestOptions): Promise<KnowledgeBase> {
        return this.http.request<KnowledgeBase>('PATCH', `/v1/knowledge-bases/${enc(id)}`, { ...opts, body });
    }

    delete(id: string, opts?: RequestOptions): Promise<void> {
        return this.http.request<void>('DELETE', `/v1/knowledge-bases/${enc(id)}`, opts);
    }
}

File: packages/sdk-dart/lib/src/resources/knowledge_bases.dart
import '../models/knowledge_base.dart';
import '../page.dart';
import '../transport.dart';

/// Knowledge bases: facts and FAQs an agent can draw on.
class KnowledgeBases {
  /// Binds the resource to a transport.
  const KnowledgeBases(this._transport);

  final Transport _transport;

  /// Lists knowledge bases.
  Future<Page<KnowledgeBase>> list(
          {int? limit, String? startingAfter, String? endingBefore}) =>
      _transport.list('/v1/knowledge-bases', KnowledgeBase.fromJson, query: {
        'limit': limit,
        'starting_after': startingAfter,
        'ending_before': endingBefore,
      });

  /// Creates one.
  Future<KnowledgeBase> create(CreateKnowledgeBase body) =>
      _transport.request('POST', '/v1/knowledge-bases', KnowledgeBase.fromJson,
          body: body.toJson());

  /// Fetches one.
  Future<KnowledgeBase> retrieve(String id) => _transport.request('GET',
      '/v1/knowledge-bases/${Uri.encodeComponent(id)}', KnowledgeBase.fromJson);

  /// Changes one. Only the fields you set are changed.
  Future<KnowledgeBase> update(String id, UpdateKnowledgeBase body) =>
      _transport.request(
          'PATCH',
          '/v1/knowledge-bases/${Uri.encodeComponent(id)}',
          KnowledgeBase.fromJson,
          body: body.toJson());

  /// Deletes one.
  Future<void> delete(String id) => _transport.requestVoid(
      'DELETE', '/v1/knowledge-bases/${Uri.encodeComponent(id)}');
}
